e2e/report-selection.spec.ts

```diff
@@ -0,0 +1,60 @@
+import { test, expect } from '@playwright/test';
+
+test('loading Vengeance DH report', async ({ page }) => {
+  await page.goto('/');
+
+  await page.getByPlaceholder('https://www.warcraftlogs.com/reports/<report code>').click();
+  await page
+    .getByPlaceholder('https://www.warcraftlogs.com/reports/<report code>')
+    .fill('https://www.warcraftlogs.com/reports/hg34bJNkWAfXCmFn');
+  await page.getByRole('heading', { name: 'Fight selection' }).waitFor();
+  await expect(page).toHaveURL('/report/hg34bJNkWAfXCmFn');
```

Comment:
When making e2e tests, I usually like to stop my test here so a single test has a single origin -> action -> result as much as possible. I am well aware this isn't ideal for most e2e tests, but in this case you're navigating over multiple completely different pages, and when that's the case I usually prefer them as separate test suites (you could have something like _player selection_, _fight selection_ and _report_ test suites). Since this test ensures the URL has to be `/report/hg34bJNkWAfXCmFn` if it succeeds, the next test can safely use `await page.goto('/report/hg34bJNkWAfXCmFn')`. The benefit is smaller, focussed tests that only test a single thing. A nice bonus is that this would also test whether opening a direct link (/report/hg34bJNkWAfXCmFn) works correctly and doesn't rely on additional state.

Reply:
I agree that there should be broken down tests as well, but I don't see any harm in having one long test that goes through the entire user journey. The only thing that might be less palatable about having both would have some level of duplication, imo.